sale_coupon_advanced/tests/test_sale_coupon_forced_reward_product.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.addons.sale_coupon.tests.common import TestSaleCouponCommon
+
+
+class TestSaleCouponForcedRewardProduct(TestSaleCouponCommon):
+    def setUp(self):
+        super().setUp()
+
+        self.env["sale.coupon.program"].search([]).write({"active": False})
+
+        # create partner for sale order.
+        self.john_doe = self.env["res.partner"].create(
+            {"name": "John Doe", "email": "john.doe@example.com", "lang": "en_US"}
+        )
+
+        self.order_forced = self.env["sale.order"].create(
+            {
+                "partner_id": self.john_doe.id,
+                "order_line": [
+                    (
+                        0,
+                        False,
+                        {
+                            "product_id": self.product_A.id,
+                            "name": "1 Product A",
+                            "product_uom": self.uom_unit.id,
+                            "product_uom_qty": 2.0,
+                        },
+                    ),
+                ],
+            }
+        )
```

Comment:
There's a `self.empty_order` attribute in TestSaleCouponCommon.
You just have to add a line.

Reply:
It's not the same partner in fact. I need to force the lang to avoid translations issue later on the process.

Reply:
The user is already created and configured. Please, check how the other test cases are made.

Reply:
Fixed